sound/soc/codecs/rt715-sdw-sdca.c

```diff
@@ -0,0 +1,470 @@
+		CH_06):
+	case RT715_SDCA_ADDR(FUNC_02, FU_AMIC_GAIN_EN, FU_DMIC_GAIN_CTRL,
+		CH_07):
+	case RT715_SDCA_ADDR(FUNC_02, FU_AMIC_GAIN_EN, FU_DMIC_GAIN_CTRL,
+		CH_08):
+	case RT715_SDCA_ADDR(FUNC_02, FU_DMIC_GAIN_EN, FU_DMIC_GAIN_CTRL,
+		CH_01):
+	case RT715_SDCA_ADDR(FUNC_02, FU_DMIC_GAIN_EN, FU_DMIC_GAIN_CTRL,
+		CH_02):
+	case RT715_SDCA_ADDR(FUNC_02, FU_DMIC_GAIN_EN, FU_DMIC_GAIN_CTRL,
+		CH_03):
+	case RT715_SDCA_ADDR(FUNC_02, FU_DMIC_GAIN_EN, FU_DMIC_GAIN_CTRL,
+		CH_04):
+	case RT715_SDCA_ADDR(FUNC_02, FU_DMIC_GAIN_EN, FU_DMIC_GAIN_CTRL,
+		CH_05):
+	case RT715_SDCA_ADDR(FUNC_02, FU_DMIC_GAIN_EN, FU_DMIC_GAIN_CTRL,
+		CH_06):
+	case RT715_SDCA_ADDR(FUNC_02, FU_DMIC_GAIN_EN, FU_DMIC_GAIN_CTRL,
+		CH_07):
+	case RT715_SDCA_ADDR(FUNC_02, FU_DMIC_GAIN_EN, FU_DMIC_GAIN_CTRL,
+		CH_08):
+		regmap_write(rt715->sdw_regmap,
+			reg | BIT(13), (val >> 8) & 0xff);
+		regmap_write(rt715->sdw_regmap, reg, val & 0xff);
+		goto _done_;
+	default:
+		break;
+	}
+
+	/* SDCA mapping case or normal address */
+	if (reg & 0x40000000 || reg < 0xffff) {
+		regmap_write(rt715->sdw_regmap, reg, val);
+	}
+
+	/* vendor registers case */
+	if (reg < 0x40000000 && reg > 0xffff) {
+		regmap_write(rt715->sdw_regmap,
+			reg | BIT(13), (val >> 8) & 0xff);
+		regmap_write(rt715->sdw_regmap, reg, val & 0xff);
+	}
+
+_done_:
+	dev_dbg(dev, "[%s] %04x <= %04x\n", __func__, reg, val);
+
+	return 0;
+}
+
+static const struct regmap_config rt715_sdca_regmap = {
+	.reg_bits = 32,
+	.val_bits = 32,
+	.readable_reg = rt715_readable_register,
+	.volatile_reg = rt715_volatile_register,
+	.max_register = 0x43ffffff,
+	.reg_defaults = rt715_reg_defaults_sdca,
+	.num_reg_defaults = ARRAY_SIZE(rt715_reg_defaults_sdca),
+	.cache_type = REGCACHE_RBTREE,
+	.use_single_read = true,
+	.use_single_write = true,
+	.reg_read = rt715_sdw_read_sdca,
+	.reg_write = rt715_sdw_write_sdca,
+};
+
+static const struct regmap_config rt715_sdca_sdw_regmap = {
+	.name = "sdw",
+	.reg_bits = 32,
+	.val_bits = 8,
+	.readable_reg = rt715_readable_register,
+	.max_register = 0x43ffffff,
+	.cache_type = REGCACHE_NONE,
+	.use_single_read = true,
+	.use_single_write = true,
+};
+
+static int rt715_update_status(struct sdw_slave *slave,
+				enum sdw_slave_status status)
+{
+	struct rt715_priv *rt715 = dev_get_drvdata(&slave->dev);
+
+	/* Update the status */
+	rt715->status = status;
+
+	/*
+	 * Perform initialization only if slave status is present and
+	 * hw_init flag is false
+	 */
+	if (rt715->hw_init || rt715->status != SDW_SLAVE_ATTACHED)
+		return 0;
+
+	/* perform I/O transfers required for Slave initialization */
+	return rt715_io_init(&slave->dev, slave);
+}
+
+static int rt715_read_prop(struct sdw_slave *slave)
+{
+	struct sdw_slave_prop *prop = &slave->prop;
+	int nval, i, num_of_ports = 1;
+	u32 bit;
+	unsigned long addr;
+	struct sdw_dpn_prop *dpn;
+
+	prop->paging_support = false;
+	if (slave->id.sdw_version == 0x03)
+		prop->paging_support = true;
```

Comment:
isn't this driver only used for SDCA? If yes, you can drop the test and set the paging support unconditionally.